main.js

```diff
@@ -1,107 +1,290 @@
-var Block = /** @class */ (function () {
-    function Block(name, width, height, image) {
+import { fields } from "./possibleImages.js";
+class Animation {
+    gameoverMessage;
+    constructor(h2Element) {
+        this.gameoverMessage = h2Element;
+    }
+    showConfetti() {
+        // Number of pieces of confetti to throw
+        const confettiCount = 200;
+        // Array to store the confetti
+        const confettiArray = [];
+        // Create the confetti pieces and add them to the array
+        for (let i = 0; i < confettiCount; i++) {
+            const confetti = document.createElement("div");
+            confetti.classList.add("confetti");
+            confetti.textContent = "🎉";
+            confetti.style.left = `${Math.random() * 100}vw`;
+            confetti.style.animationDuration = `${Math.random() * 2 + 3}s`;
+            confettiArray.push(confetti);
+        }
+        // Function to start the confetti animation
+        const startConfetti = () => {
+            if (this.gameoverMessage) {
+                this.gameoverMessage.textContent = "You Won!";
+            }
+            confettiArray.forEach((confetti) => {
+                document.body.appendChild(confetti);
+            });
+        };
+        // Function to stop the confetti animation
+        const stopConfetti = () => {
+            confettiArray.forEach((confetti) => {
+                confetti.remove();
+            });
+        };
+        // Start the confetti animation
+        startConfetti();
+        // Stop the confetti animation after 5 seconds
+        setTimeout(stopConfetti, 5000);
+    }
+}
+class ElementCreator {
+    name;
+    width;
+    height;
+    image;
+    className;
+    style;
+    constructor(name, width, height, image, style, className) {
+        this.name = name;
+        this.width = width;
+        this.height = height;
+        if (image) {
+            this.image = image;
+        }
+        if (style) {
+            this.style = style;
+        }
+        if (className) {
+            this.className = className;
+        }
+    }
+    createDiv() {
+        const div = document.createElement("div");
+        div.style.width = this.width + "px";
+        div.style.height = this.height + "px";
+        if (this.className) {
+            div.classList.add(this.className ? this.className : "");
+        }
+        if (this.style) {
+            Object.assign(div.style, this.style);
+        }
+        return div;
+    }
+    createImg() {
+        const img = document.createElement("img");
+        img.src = this.image ? this.image : "";
+        img.style.width = this.width + "px";
+        img.style.height = this.height + "px";
+        if (this.className) {
+            img.classList.add(this.className ? this.className : "");
+        }
+        if (this.style) {
+            Object.assign(img.style, this.style);
+        }
+        return img;
+    }
+    createElement() {
```

Comment:
I don't like how you managed the createElement class, the purpose of the element creator class is the create the element by not calling the createElement() function, can you think of another solution here?
Tip: Use the createElement function to create an instance of the ElementCreator class, then you will have getElement which will return you the generated element.